controllers/tasks.js

```diff
@@ -126,21 +126,36 @@ const createTask = [
     const spaceIds = spaceTree.map((space) => space.id);
     const actionsCursor = fetchActions(actionsCollection, spaceIds);
 
-    // fetch users by space ids; every space with users has a 'memberships' array
-    // for each user, add 'additional info' by querying 'users' collection
-    // convert user "type" value to be either 'light' or 'graasp'
-    const usersArray = await fetchUsers(itemsCollection, spaceIds);
-    const usersWithInfo = [];
-    for (let i = 0; i < usersArray.length; i += 1) {
-      // eslint-disable-next-line no-await-in-loop
-      const userWithInfo = await appendUserInfo(usersCollection, usersArray[i]);
-      usersWithInfo.push(userWithInfo);
-    }
+    // fetch users by space ids; convert mongo cursor to array
+    const usersCursor = fetchUsers(itemsCollection, spaceIds);
+    const usersArray = await usersCursor.toArray();
+
+    // 'usersArray' is an array of objects, each with a 'memberships' key
+    // each 'memberships' key holds an array of objects, each of the format { userId: <mongoId> }
+    // i.e. usersArray = [ { memberships: [ { userId: <mongoId> }, ... ] }, ... ]
+    let userIds = [];
+    usersArray.forEach(({ memberships }) => {
+      if (memberships) {
+        memberships.forEach(({ userId }) => {
+          userIds.push(userId.toString());
+        });
+      }
+    });
```

Comment:
another way of doing this:

```javascript
let userIds = usersArray.reduce((acc, { memberships = [] }) =>
  acc.concat(memberships.map(({ userId }) => userId.toString())),
[]);
```

Reply:
I tested this function versus the two `.forEach()` loops and it is **significantly** slower. E.g. for a `usersArray` of length 50000, this function takes ~24 seconds to complete, while for the same `usersArray` the two `.forEach()` loops complete in <100ms. I also find the `.forEach()` syntax easier to read. So I will keep as is.

Reply:
The slowness I didn't know. You can use the online took jsperf to test those. It's probably even faster with simple for()'s
The readability, I agree.